syntax = "proto3";

package transaction_verification;

message User {
    string name = 1;
    string contact = 2;
}

message CreditCard {
    string number = 1;
    string expirationDate = 2;
    string cvv = 3;
}

message TransactionVerificationRequest {
    User user = 1;
    CreditCard creditCard = 2;
}

message TransactionVerificationResponse {
    bool is_valid = 1;
    string message = 2;
}

service TransactionVerificationService {
    rpc VerifyUser(TransactionVerificationRequest) returns (TransactionVerificationResponse) {}
    rpc VerifyCreditCard(TransactionVerificationRequest) returns (TransactionVerificationResponse) {}
    rpc VerifyCreditCardInvalid(TransactionVerificationRequest) returns (TransactionVerificationResponse) {}
}